src/isolate.rs

```diff
@@ -152,14 +152,27 @@ pub type PrepareStackTraceCallback<'s> = extern "C" fn(
   Local<'s, Array>,
 ) -> *mut *const Value;
 
-// System V ABI: MaybeLocal<Value> returned in a register.
-#[cfg(not(target_os = "windows"))]
+// System V AMD64 ABI: MaybeLocal<Value> returned in a register.
+#[cfg(all(not(target_os = "windows"), target_pointer_width = "64"))]
 pub type PrepareStackTraceCallback<'s> = extern "C" fn(
   Local<'s, Context>,
   Local<'s, Value>,
   Local<'s, Array>,
 ) -> *const Value;
 
+// System V i386 ABI: Local<Value> returned in hidden pointer (struct).
+#[cfg(all(not(target_os = "windows"), target_pointer_width = "32"))]
+#[repr(C)]
+pub struct PrepareStackTraceCallbackRet(*const Value);
+
+#[cfg(all(not(target_os = "windows"), target_pointer_width = "32"))]
+pub type PrepareStackTraceCallback<'s> =
+  extern "C" fn(
+    Local<'s, Context>,
+    Local<'s, Value>,
+    Local<'s, Array>,
+  ) -> PrepareStackTraceCallbackRet;
+
```

Comment:
I think it should be possible to use PrepareStackTraceCallbackRet for both 32 and 64 bits. Word-sized structs are returned in registers so under the hood it compiles down to the same thing.